siliconcompiler/tools/yosys/syn_asic.py

```diff
@@ -90,6 +84,16 @@ def setup_asic(chip):
                  ",".join(['library', mainlib, 'option', 'var', 'yosys_buffer_output']),
                  step=step, index=index)
 
+    chip.set('tool', tool, 'task', task, 'var', 'map_adders',
+             "true" if chip.valid('library', mainlib, 'option', 'file', 'yosys_addermap') else
```

Comment:
You might also want to check that the addermap got set for the particular step/index beyond just the keypath being valid, otherwise I could see this behaving funny if it the adder map is provided for one Yosys node but not another.